Azure AI model inference data-plane API

santiagxf opened the pull request:
# Data Plane API Specification Update Pull Request

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.


The Azure AI model inference API allow customers to consume model inference from a variety of models deployed in Azure AI Foundry. It's supported in Azure AI model inference, Serverless API Endpoints, and Managed Online Endpoints. This PR introduces the data plane contracts used in the API.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## API Info: The Basics

Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Schedule a data plane REST API spec review](https://aka.ms/azsdk/onboarding/restapischedule)

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories) 
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

santiagxf committed:
init

santiagxf committed:
fix: examples

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Common](https://apiview.dev/Assemblies/Review/c9b24f9b09434660a3db723f9d8efe07?revisionId=a2840ef6ed1f4a04a31f0a90874b088e)

santiagxf committed:
fix: tsp

santiagxf committed:
fix: ci

santiagxf committed:
fix: release

santiagxf committed:
fix: ci

santiagxf committed:
fix: ci

santiagxf committed:
fix: ci

dargilco commented on specification/ai/ModelInference/models/common.tsp:
```diff
@@ -0,0 +1,54 @@
+import "@typespec/rest";
```
I can't fine `response_format` here (or any .tsp file). But I do see it in the GetChatCompletions_Maximum example.

santiagxf replied:
It's here: https://github.com/Azure/azure-rest-api-specs/raw/d579b4744aa86f1eb26d491ff2f7f3d5d753b150/specification/ai/ModelInference/models/chat_completions.tsp#L94

danieljurek committed:
Removes azure-resource-manager-schemas from SDK automation. It can be re-added without reverting this commit. (#30893) (#31918)

This change produces an extra warning but no obvious new errors.

danieljurek committed:
Seed package-lock.json for spell check (#32032)

* Seed package-lock.json for spell check

* Update cSpell.json to use v0.2 config

* exclude checking the spell check configuration

santiagxf committed:
fix

ChenxiJiang333 committed:
[Compute] Update readme.python.md (#32016)

* Update readme.md

* Update readme.python.md

* Fix tag name in readme.md

* Remove outdated package tags from readme.python.md

* Remove outdated package tags from readme.python.md

* update readme.python.md

* update readme.python.md

* fix cspell

* fix cspell

* update readme.python.md

* update readme.python.md

---------

Co-authored-by: Yuchao Yan <yuchaoyan@microsoft.com>
Co-authored-by: msyyc <70930885+msyyc@users.noreply.github.com>

SethuramS committed:
Add list of allowed classifications in description for Maintenance Configuration (#31941)

Co-authored-by: Sethuram S <sethurams@microsoft.com>

msyyc committed:
[compute] Update readme.python.md (#32036)

Jingshu918 committed:
[Synapse]Sync IR Features From DataFactory (#31774)

* [Synapse]Sync IR Features From DataFactory

* fix

* fix

---------

Co-authored-by: Frey-Wang <binluwang@microsoft.com>

santiagxf committed:
fix

santiagxf committed:
fix: spell

santiagxf committed:
fix: ci

santiagxf committed:
fix: words

brrusino committed:
Microsoft.Chaos TypeSpec migration (#31664)

* sorted existing swagger

* initial autogenerated typespec

* initial TypeSpec updates to resolve warnings

* added generated files

* changed inputs to typespec-generated openapi

* Revert "added generated files"

This reverts commit c09b60a602d003f22606a64d26dd4e077963038d.

Revert "initial TypeSpec updates to resolve warnings"

This reverts commit 805007d2f9a078a8553853b638b246f1580dab07.

* added new typespec migration api version

* file format updates

* reverted back unintentionally changed spec from previous api version

* various modelvalidation fixes

* fix

* model validation fixes

* linting fixes

* fixes for linting

* fixing swagger avocado

* fixed x-ms-example for privatelink list

* updated readme

* fixed tspconfig formatting

* updated tspconfig

* updates to try to fix breaking sdk changes

* trying to fix typescript client generation

* updated tspconfig

* another attempted fix for typescript sdk

* updates to client.tsp

* attempted ts fix

* updated tspconfig

* changed tspconfig back to original values

* update to tspconfig for ts sdk

* reverted back tspconfig

* attempted javascript replacement for SDK operation

* LintDiff fix

* LintDiff fix

* updated client.tsp

* updated client.tsp for csharp replacements

* updating metadata types for csharp client

* updated csharp client replacements

* updating target location visibility

* updates to fix net sdk breaking changes

* updated for net sdk

* fixing net sdk failures

* fixed client.tsp format

* updated visibility for readonly properties

* updated managed identity model reference

* updated readme directives based on feedback

* breaking change fix

* split up models into separate files

* changed file casing

* fixed bad file reference

* fixed spelling failure

* fixed tsp formatting

mikeharder committed:
[cSpell.json] Add "eng/**" to ignorePaths (#32044)

hmmorales committed:
Handle C# SDK Naming Requirements for Autogenerated Specs (#32006)

* test

* update client.tsp file

* ...

* ...

* ci fix

* ...

* rename dataflow prop & remove workaround

andpiccione committed:
[Microsoft.ConfidentialLedger] Add new ARM API version preview/2024-09-19-preview (#30653)

* Copy files from preview/2024-07-09-preview

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to preview/2024-09-19-preview

Updated the API version from preview/2024-07-09-preview to preview/2024-09-19-preview.

* Added tag for 2024-09-19-preview in readme file

* Add enclave platform property

* Fix prettier

* Add application type property

JackTn committed:
[SDK Suppressions labels] Add new action to update suppressions labels (#31878)

* initial version

* use variables to be used in next steps

* removed unused code and test branch

* added function description

* updated dependencies

* removed test branch

* add comment to to explain this behavior

* Add trailing newline

---------

Co-authored-by: Mike Harder <mharder@microsoft.com>
Co-authored-by: Ray Chen <raychen@microsoft.com>

dependabot[bot] committed:
Bump the typespec group across 1 directory with 2 updates (#32051)

Bumps the typespec group with 2 updates in the / directory: [@azure-tools/typespec-apiview](https://github.com/Azure/azure-sdk-tools) and [@typespec/rest](https://github.com/microsoft/typespec).


Updates `@azure-tools/typespec-apiview` from 0.4.9 to 0.5.0
- [Release notes](https://github.com/Azure/azure-sdk-tools/releases)
- [Commits](https://github.com/Azure/azure-sdk-tools/commits/doc-warden_0.5.0)

Updates `@typespec/rest` from 0.63.0 to 0.63.1
- [Release notes](https://github.com/microsoft/typespec/releases)
- [Commits](https://github.com/microsoft/typespec/compare/typespec@0.63.0...@typespec/rest@0.63.1)

---
updated-dependencies:
- dependency-name: "@azure-tools/typespec-apiview"
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: typespec
- dependency-name: "@typespec/rest"
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: typespec
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump oav from 3.5.1 to 3.6.0 (#32052)

Bumps [oav](https://github.com/azure/oav) from 3.5.1 to 3.6.0.
- [Release notes](https://github.com/azure/oav/releases)
- [Changelog](https://github.com/Azure/oav/blob/develop/ChangeLog.md)
- [Commits](https://github.com/azure/oav/commits)

---
updated-dependencies:
- dependency-name: oav
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

hiaga committed:
[recoveryservices] Microsoft.RecoveryServices adding version 2024-09-30-preview  (#32041)

* Microsoft.RecoveryServices adding version 2024-09-30-preview (#30405)

* Copy files from stable/2024-04-01

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to preview/2024-09-30-preview

Updated the API version from stable/2024-04-01 to preview/2024-09-30-preview.

* Added tag for 2024-09-30-preview in readme file

* resolve readme conflict (#31990)

added latest stable folder

Co-authored-by: Himanshu Agarwal <hiaga@microsoft.com>

---------

Co-authored-by: DevBaburaj <devbabu@microsoft.com>
Co-authored-by: justbelieve <ankur@microsoft.com>

chantalolieman committed:
bugfix in api spec (#31695)

scottaddie committed:
Rebranding of Azure AI Studio (#32063)

* Rebranding of Azure AI Studio

* Fix spelling error

xuhumsft committed:
Dev sentinel securityinsights microsoft.security insights 2024 10 01 preview (#31790)

* Copy files from preview/2024-04-01-preview

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to preview/2024-10-01-preview

Updated the API version from preview/2024-04-01-preview to preview/2024-10-01-preview.

* Added tag for 2024-10-01-preview in readme file

* support odata search in listing product packages (#31740)

* support odata search in listing product packages

* add new supported contentKind

* fix perttier errors

* Fix Data connectors API Typos (#31806)

* Add PurviewAudit connector (#31821)

* Add PurviewAudit connector

* prettier fix

* prettier fix

---------

Co-authored-by: Will King <kingwil@users.noreply.github.com>

* resolve comments and fix the prettier issue (#31965)

* Watchlist fix (#32047)

* reorder TI parameter to address breaking change errors (#32053)

---------

Co-authored-by: NISHILBISHT <nishilbisht@microsoft.com>
Co-authored-by: sagamzu <52034287+sagamzu@users.noreply.github.com>
Co-authored-by: kingwil <5839248+kingwil@users.noreply.github.com>
Co-authored-by: Will King <kingwil@users.noreply.github.com>
Co-authored-by: nbatyrbb <115049492+nbatyrbb@users.noreply.github.com>

danieljurek committed:
Don't check custom-words.txt (#32065)

* ignore custom-words.txt

* test change to custom-words.txt

* Revert test change to custom-words.txt

gigigoo0 committed:
Fix get polygon response schema (#32061)

koyasu221b committed:
Route 2024-07-01-preview update for RouteRange (#32086)

* Update isSimplifiedPolygon

* Fix cspell

---------

Co-authored-by: Will Huang <huangwill@microsoft.com>

weidongxu-microsoft committed:
java, OpenAI.Inference, migrate to clientNamespace (#31643)

dpwatrous committed:
[Batch] Normalize mgmt plane doc links (#32069)

Fixed some broken links, and normalized all doc URLs to
learn.microsoft.com without the en-us locale.

dpwatrous committed:
[Batch] Normalize data plane doc links (#32070)

Fixed some broken links, and normalized all doc URLs to
learn.microsoft.com without the en-us locale.

SpencerOfwiti committed:
Update examples and descriptions and update ApiKeyNameParameter for Nginx swagger (#32042)

* Update examples and descriptions for Nginx swagger

We recently merged this PR, [Nginx 2024 11 01 preview #31409](https://github.com/Azure/azure-rest-api-specs/pull/31409), but we found out some examples and description were not desired. we just want to update these and have no functional changes

* Add missing periods in JSON descriptions.

* Add x-ms-parameter-location to apiKeyName

Add `"x-ms-parameter-location": "method"` property to `ApiKeyNameParameter` for easier AutoRest generation.

chidozieononiwu committed:
Update Source and Target Commit for APIView TypeSpec generation (#32076)

fangru2023 committed:
Fix the api descrption (#31188)

* decription fix.

* remove \n

* fix invalid armId.

* fix \n

* fix arm id.

* Create sdk-suppressions.yaml

* Update specification/standbypool/StandbyPool.Management/sdk-suppressions.yaml

* edit cSpell.json.

* edit cSpell.json.

---------

Co-authored-by: kazrael2119 <98569699+kazrael2119@users.noreply.github.com>

Renyx1219 committed:
Release Microsoft.DatabaseWatcher GA Version stable/2025-01-02 (#32028)

* Add stable version 2025-01-02.

* Add examples for new API version.

* Update readme.md

* Add stable version 2025-01-02.

* Add examples for new API version.

* Update readme.md

* Fix TypeSpec Validation Error.

* Revert "Fix TypeSpec Validation Error."

This reverts commit 82c73af01bc09d343234b87c7ebd5b4cd1672d1a.

* Re-compile the code.

* Use eastus2 as example region.

* Update doc strings for consistence.

* Fix SpellCheck issue.

* Remove unused SqlVmTargetProperties in GA version.

---------

Co-authored-by: Yixiong Ren <yixiongren@microsoft.com>
Co-authored-by: Shubham <72555762+Shubham-9999-bot@users.noreply.github.com>

l0lawrence committed:
[eg] make req/optional more explicit in readme (#32079)

* update

* update for clarity

mjmadhu committed:
PlaywrightTesting -Bringing common model in shared namespace (#32019)

* Created PlaywrightTesting.Shared namespace

* tsp format fixed

santiagxf committed:
fix: ci

santiagxf committed:
Merge branch 'main' into add-models

FRL200 committed:
Update default values and descriptions for Key Vault on API Version 2024-11-01 (#31713)

* Copy files from stable/2023-07-01

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to stable/2024-11-01

Updated the API version from stable/2023-07-01 to stable/2024-11-01.

* Added tag for 2024-11-01 in readme file

* Removed unnecessary folder

* Update default values and descriptions for Key Vault (#31683)

* Initial Changes to Spec

* Prettier ran

* Ran prettier

---------

Co-authored-by: Fernando Rauseo Landa <frauseolanda@microsoft.com>

* Ran prettier

* Ran prettier on all files

* Update managedHsm.json (#31732)

* typo fix

---------

Co-authored-by: Fernando Rauseo Landa <frauseolanda@microsoft.com>
Co-authored-by: Daniel Cherrett <49488952+Cherrett@users.noreply.github.com>

FRL200 committed:
Release keyvault microsoft.key vault preview/2024 12 01 preview (#31796)

* Copy files from preview/2024-04-01-preview

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to preview/2024-12-01-preview

Updated the API version from preview/2024-04-01-preview to preview/2024-12-01-preview.

* Added tag for 2024-12-01-preview in readme file

* Added Azure Managed HSM C SKUs to preview version (#31733)

* Update managedHsm.json

* Update managedHsm.json

* Changes to update default values and description and prettier ran (#31718)

Co-authored-by: Fernando Rauseo Landa <frauseolanda@microsoft.com>

* Update readme

* misc fix

* update tag

* resolve ci issue

* fixed typos and override words

---------

Co-authored-by: Fernando Rauseo Landa <frauseolanda@microsoft.com>
Co-authored-by: Daniel Cherrett <49488952+Cherrett@users.noreply.github.com>
Co-authored-by: Chenjie Shi <tadelesh.shi@live.cn>

santiagxf committed:
fix: ci

santiagxf committed:
Merge branch 'add-models' of https://github.com/santiagxf/azure-rest-api-specs into add-models

ChenxiJiang333 committed:
[Container Service] Update readme.python.md (#32112)

* Update readme.python.md

* fix cspell

---------

Co-authored-by: Yuchao Yan <yuchaoyan@microsoft.com>

JackTn committed:
[BUG] Github Action For Sdk Suppressions (#32124)

* add filter suppression

* parse changefiles

santiagxf committed:
fix: exception

santiagxf committed:
fix: exception

santiagxf committed:
fix: ci

santiagxf committed:
fix: ci

santiagxf committed:
Merge branch 'main' into add-models

danieljurek committed:
Spelling: Allow compound words, fix JSON syntax (#32140)

azure-sdk committed:
remove erroneous package inclusion during package-properties evaluation (#32129)

Co-authored-by: Scott Beddall <scbedd@microsoft.com>

santiagxf committed:
Merge branch 'main' into add-models

santiagxf committed:
fix: ci

santiagxf committed:
fix: ci

santiagxf committed:
Merge branch 'add-models' of https://github.com/santiagxf/azure-rest-api-specs into add-models

santiagxf committed:
Merge branch 'add-models' of https://github.com/santiagxf/azure-rest-api-specs into add-models

santiagxf committed:
clean
